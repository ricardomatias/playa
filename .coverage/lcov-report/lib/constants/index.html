<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for lib/constants</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> lib/constants
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">87.5% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>56/64</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">88.89% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>56/63</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="chords.js"><a href="chords.js.html">chords.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="36" class="abs high">36/36</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="36" class="abs high">36/36</td>
	</tr>

<tr>
	<td class="file empty" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file high" data-value="intervals.js"><a href="intervals.js.html">intervals.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file high" data-value="note.js"><a href="note.js.html">note.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file high" data-value="rhythmsDistributions.js"><a href="rhythmsDistributions.js.html">rhythmsDistributions.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file high" data-value="scales.js"><a href="scales.js.html">scales.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

<tr>
	<td class="file medium" data-value="ticks.js"><a href="ticks.js.html">ticks.js</a></td>
	<td data-value="60" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 60%;"></div><div class="cover-empty" style="width:40%;"></div></div></td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="20" class="abs medium">12/20</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="6" class="abs medium">4/6</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="63.16" class="pct medium">63.16%</td>
	<td data-value="19" class="abs medium">12/19</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Feb 20 2019 14:43:58 GMT+0100 (Central European Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
