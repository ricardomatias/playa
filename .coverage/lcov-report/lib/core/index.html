<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for lib/core</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> lib/core
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">88.17% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>343/389</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">81.98% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>141/172</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.82% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>57/68</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">87.83% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>332/378</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="Chord.js"><a href="Chord.js.html">Chord.js</a></td>
	<td data-value="96.9" class="pic high"><div class="chart"><div class="cover-fill" style="width: 96%;"></div><div class="cover-empty" style="width:4%;"></div></div></td>
	<td data-value="96.9" class="pct high">96.9%</td>
	<td data-value="129" class="abs high">125/129</td>
	<td data-value="90.91" class="pct high">90.91%</td>
	<td data-value="44" class="abs high">40/44</td>
	<td data-value="94.44" class="pct high">94.44%</td>
	<td data-value="18" class="abs high">17/18</td>
	<td data-value="96.77" class="pct high">96.77%</td>
	<td data-value="124" class="abs high">120/124</td>
	</tr>

<tr>
	<td class="file high" data-value="HarmonyBase.js"><a href="HarmonyBase.js.html">HarmonyBase.js</a></td>
	<td data-value="84.62" class="pic high"><div class="chart"><div class="cover-fill" style="width: 84%;"></div><div class="cover-empty" style="width:16%;"></div></div></td>
	<td data-value="84.62" class="pct high">84.62%</td>
	<td data-value="13" class="abs high">11/13</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="6" class="abs medium">4/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="84.62" class="pct high">84.62%</td>
	<td data-value="13" class="abs high">11/13</td>
	</tr>

<tr>
	<td class="file high" data-value="Key.js"><a href="Key.js.html">Key.js</a></td>
	<td data-value="94.67" class="pic high"><div class="chart"><div class="cover-fill" style="width: 94%;"></div><div class="cover-empty" style="width:6%;"></div></div></td>
	<td data-value="94.67" class="pct high">94.67%</td>
	<td data-value="75" class="abs high">71/75</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="36" class="abs high">30/36</td>
	<td data-value="92.31" class="pct high">92.31%</td>
	<td data-value="13" class="abs high">12/13</td>
	<td data-value="94.67" class="pct high">94.67%</td>
	<td data-value="75" class="abs high">71/75</td>
	</tr>

<tr>
	<td class="file low" data-value="Mode.js"><a href="Mode.js.html">Mode.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="24" class="abs low">0/24</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="24" class="abs low">0/24</td>
	</tr>

<tr>
	<td class="file high" data-value="Note.js"><a href="Note.js.html">Note.js</a></td>
	<td data-value="89.74" class="pic high"><div class="chart"><div class="cover-fill" style="width: 89%;"></div><div class="cover-empty" style="width:11%;"></div></div></td>
	<td data-value="89.74" class="pct high">89.74%</td>
	<td data-value="78" class="abs high">70/78</td>
	<td data-value="90.48" class="pct high">90.48%</td>
	<td data-value="42" class="abs high">38/42</td>
	<td data-value="84.21" class="pct high">84.21%</td>
	<td data-value="19" class="abs high">16/19</td>
	<td data-value="89.61" class="pct high">89.61%</td>
	<td data-value="77" class="abs high">69/77</td>
	</tr>

<tr>
	<td class="file high" data-value="Scale.js"><a href="Scale.js.html">Scale.js</a></td>
	<td data-value="93.94" class="pic high"><div class="chart"><div class="cover-fill" style="width: 93%;"></div><div class="cover-empty" style="width:7%;"></div></div></td>
	<td data-value="93.94" class="pct high">93.94%</td>
	<td data-value="66" class="abs high">62/66</td>
	<td data-value="82.86" class="pct high">82.86%</td>
	<td data-value="35" class="abs high">29/35</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="10" class="abs high">8/10</td>
	<td data-value="93.44" class="pct high">93.44%</td>
	<td data-value="61" class="abs high">57/61</td>
	</tr>

<tr>
	<td class="file empty" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file high" data-value="types.js"><a href="types.js.html">types.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Feb 20 2019 15:09:45 GMT+0100 (Central European Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
