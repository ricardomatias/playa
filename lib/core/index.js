/**
* @namespace {module} Core
*/
export { Note } from './Note';
export { default as Chord } from './Chord';
export { default as Scale } from './Scale';
export { default as Key } from './Key';
export { Time } from './Time';
export { Phrase } from './Phrase';
export { PianoRoll } from './PianoRoll';

