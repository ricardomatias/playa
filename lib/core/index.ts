export * from './Note';
export * from './Chord';
export * from './Scale';
export * from './Key';
export * from './Time';

export * from './Event';
export * from './ChordEvent';
export * from './NoteEvent';
