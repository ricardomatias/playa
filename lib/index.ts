// export * from './constants';
// export * from './composition';
// export * from './tools';
// export * from './analysis';

export { Note } from './core/Note';
// export { Chord } from './core/Chord';
export { Scale } from './core/Scale';
export { Key } from './core/Key';
// export * from './core/Time';

// export * from './core/Event';
// export * from './core/ChordEvent';
// export * from './core/NoteEvent';
