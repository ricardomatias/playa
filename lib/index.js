import * as Constants from './constants';
import * as Core from './core';
import * as Functional from './functional';
import * as Tools from './tools';

export {
	Core,
	Constants,
	Functional,
	Tools,
};
