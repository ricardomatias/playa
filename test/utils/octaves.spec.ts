import * as R from 'ramda';
import assignOctaves from '../../lib/utils/octaves';
import { Scale, Chord, Note } from '../../lib/core';

const toNoteObj = (note: Note) => ({ [note.n]: note.m });

describe('An Octaves Test Suite', () => {
	it('should map - chromatic', () => {
		const scale = new Scale('C', Scale.Intervals.Chromatic);
		const octaves = R.map(toNoteObj, assignOctaves(scale.notes));

		expect(octaves).toMatchSnapshot();
	});

	it('should map - natural', () => {
		const scale = new Scale('C', Scale.Major);
		const octaves = R.map(toNoteObj, assignOctaves(scale.notes));

		expect(octaves).toHaveLength(75);
		expect(octaves).toMatchSnapshot();
	});

	it('should map - sharp', () => {
		const scale = new Scale('E', Scale.Major);
		const octaves = R.map(toNoteObj, assignOctaves(scale.notes, [ 3, 2 ]));

		expect(octaves).toHaveLength(14);
		expect(octaves).toMatchSnapshot();
	});

	it('should map - flat', () => {
		const scale = new Scale('Ab', Scale.Major);
		const octaves = R.map(toNoteObj, assignOctaves(scale.notes, [ 4, 4 ]));

		expect(octaves).toHaveLength(28);
		expect(octaves).toMatchSnapshot();
	});

	it('should map - pentatonic', () => {
		const scale = new Scale('Eb', Scale.Intervals.MajorPentatonic);
		const octaves = R.map(toNoteObj, assignOctaves(scale.notes, [ -2, 1 ]));

		expect(octaves).toHaveLength(5);
		expect(octaves).toMatchSnapshot();
	});

	it('should map - major with flats', () => {
		const scale = new Scale('F', Scale.Major);
		const octaves = R.map(toNoteObj, assignOctaves(scale.notes, [ -1, 1 ], {
			type: 'scale', hasFlats: true,
		}));

		expect(octaves).toHaveLength(7);
		expect(octaves).toMatchSnapshot();
	});

	it('should map 9m chord', () => {
		const chord = Chord.fromName('Abm9');
		const octaves = R.map(toNoteObj, assignOctaves(chord.notes, [ 4, 1 ], { type: 'chord' }));

		expect(octaves).toHaveLength(5);
		expect(octaves).toMatchSnapshot();
	});

	it('should map M13 chord', () => {
		const chord = Chord.fromName('CM13');
		const octaves = R.map(toNoteObj, assignOctaves(chord.notes, [ 4, 2 ], { type: 'chord' }));

		expect(octaves).toHaveLength(12);
		expect(octaves).toMatchSnapshot();
	});
});

