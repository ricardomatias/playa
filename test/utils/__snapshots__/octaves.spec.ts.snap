// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`An Octaves Test Suite should map - chromatic 1`] = `
Array [
  Object {
    "C-2": 0,
  },
  Object {
    "C#-2": 1,
  },
  Object {
    "D-2": 2,
  },
  Object {
    "D#-2": 3,
  },
  Object {
    "E-2": 4,
  },
  Object {
    "F-2": 5,
  },
  Object {
    "F#-2": 6,
  },
  Object {
    "G-2": 7,
  },
  Object {
    "G#-2": 8,
  },
  Object {
    "A-2": 9,
  },
  Object {
    "A#-2": 10,
  },
  Object {
    "B-2": 11,
  },
  Object {
    "C-1": 12,
  },
  Object {
    "C#-1": 13,
  },
  Object {
    "D-1": 14,
  },
  Object {
    "D#-1": 15,
  },
  Object {
    "E-1": 16,
  },
  Object {
    "F-1": 17,
  },
  Object {
    "F#-1": 18,
  },
  Object {
    "G-1": 19,
  },
  Object {
    "G#-1": 20,
  },
  Object {
    "A-1": 21,
  },
  Object {
    "A#-1": 22,
  },
  Object {
    "B-1": 23,
  },
  Object {
    "C0": 24,
  },
  Object {
    "C#0": 25,
  },
  Object {
    "D0": 26,
  },
  Object {
    "D#0": 27,
  },
  Object {
    "E0": 28,
  },
  Object {
    "F0": 29,
  },
  Object {
    "F#0": 30,
  },
  Object {
    "G0": 31,
  },
  Object {
    "G#0": 32,
  },
  Object {
    "A0": 33,
  },
  Object {
    "A#0": 34,
  },
  Object {
    "B0": 35,
  },
  Object {
    "C1": 36,
  },
  Object {
    "C#1": 37,
  },
  Object {
    "D1": 38,
  },
  Object {
    "D#1": 39,
  },
  Object {
    "E1": 40,
  },
  Object {
    "F1": 41,
  },
  Object {
    "F#1": 42,
  },
  Object {
    "G1": 43,
  },
  Object {
    "G#1": 44,
  },
  Object {
    "A1": 45,
  },
  Object {
    "A#1": 46,
  },
  Object {
    "B1": 47,
  },
  Object {
    "C2": 48,
  },
  Object {
    "C#2": 49,
  },
  Object {
    "D2": 50,
  },
  Object {
    "D#2": 51,
  },
  Object {
    "E2": 52,
  },
  Object {
    "F2": 53,
  },
  Object {
    "F#2": 54,
  },
  Object {
    "G2": 55,
  },
  Object {
    "G#2": 56,
  },
  Object {
    "A2": 57,
  },
  Object {
    "A#2": 58,
  },
  Object {
    "B2": 59,
  },
  Object {
    "C3": 60,
  },
  Object {
    "C#3": 61,
  },
  Object {
    "D3": 62,
  },
  Object {
    "D#3": 63,
  },
  Object {
    "E3": 64,
  },
  Object {
    "F3": 65,
  },
  Object {
    "F#3": 66,
  },
  Object {
    "G3": 67,
  },
  Object {
    "G#3": 68,
  },
  Object {
    "A3": 69,
  },
  Object {
    "A#3": 70,
  },
  Object {
    "B3": 71,
  },
  Object {
    "C4": 72,
  },
  Object {
    "C#4": 73,
  },
  Object {
    "D4": 74,
  },
  Object {
    "D#4": 75,
  },
  Object {
    "E4": 76,
  },
  Object {
    "F4": 77,
  },
  Object {
    "F#4": 78,
  },
  Object {
    "G4": 79,
  },
  Object {
    "G#4": 80,
  },
  Object {
    "A4": 81,
  },
  Object {
    "A#4": 82,
  },
  Object {
    "B4": 83,
  },
  Object {
    "C5": 84,
  },
  Object {
    "C#5": 85,
  },
  Object {
    "D5": 86,
  },
  Object {
    "D#5": 87,
  },
  Object {
    "E5": 88,
  },
  Object {
    "F5": 89,
  },
  Object {
    "F#5": 90,
  },
  Object {
    "G5": 91,
  },
  Object {
    "G#5": 92,
  },
  Object {
    "A5": 93,
  },
  Object {
    "A#5": 94,
  },
  Object {
    "B5": 95,
  },
  Object {
    "C6": 96,
  },
  Object {
    "C#6": 97,
  },
  Object {
    "D6": 98,
  },
  Object {
    "D#6": 99,
  },
  Object {
    "E6": 100,
  },
  Object {
    "F6": 101,
  },
  Object {
    "F#6": 102,
  },
  Object {
    "G6": 103,
  },
  Object {
    "G#6": 104,
  },
  Object {
    "A6": 105,
  },
  Object {
    "A#6": 106,
  },
  Object {
    "B6": 107,
  },
  Object {
    "C7": 108,
  },
  Object {
    "C#7": 109,
  },
  Object {
    "D7": 110,
  },
  Object {
    "D#7": 111,
  },
  Object {
    "E7": 112,
  },
  Object {
    "F7": 113,
  },
  Object {
    "F#7": 114,
  },
  Object {
    "G7": 115,
  },
  Object {
    "G#7": 116,
  },
  Object {
    "A7": 117,
  },
  Object {
    "A#7": 118,
  },
  Object {
    "B7": 119,
  },
  Object {
    "C8": 120,
  },
  Object {
    "C#8": 121,
  },
  Object {
    "D8": 122,
  },
  Object {
    "D#8": 123,
  },
  Object {
    "E8": 124,
  },
  Object {
    "F8": 125,
  },
  Object {
    "F#8": 126,
  },
  Object {
    "G8": 127,
  },
]
`;

exports[`An Octaves Test Suite should map - flat 1`] = `
Array [
  Object {
    "Ab4": 80,
  },
  Object {
    "Bb4": 82,
  },
  Object {
    "C5": 84,
  },
  Object {
    "Db5": 85,
  },
  Object {
    "Eb5": 87,
  },
  Object {
    "F5": 89,
  },
  Object {
    "G5": 91,
  },
  Object {
    "Ab5": 92,
  },
  Object {
    "Bb5": 94,
  },
  Object {
    "C6": 96,
  },
  Object {
    "Db6": 97,
  },
  Object {
    "Eb6": 99,
  },
  Object {
    "F6": 101,
  },
  Object {
    "G6": 103,
  },
  Object {
    "Ab6": 104,
  },
  Object {
    "Bb6": 106,
  },
  Object {
    "C7": 108,
  },
  Object {
    "Db7": 109,
  },
  Object {
    "Eb7": 111,
  },
  Object {
    "F7": 113,
  },
  Object {
    "G7": 115,
  },
  Object {
    "Ab7": 116,
  },
  Object {
    "Bb7": 118,
  },
  Object {
    "C8": 120,
  },
  Object {
    "Db8": 121,
  },
  Object {
    "Eb8": 123,
  },
  Object {
    "F8": 125,
  },
  Object {
    "G8": 127,
  },
]
`;

exports[`An Octaves Test Suite should map - major with flats 1`] = `
Array [
  Object {
    "F-1": 17,
  },
  Object {
    "G-1": 19,
  },
  Object {
    "A-1": 21,
  },
  Object {
    "Bb-1": 22,
  },
  Object {
    "C0": 24,
  },
  Object {
    "D0": 26,
  },
  Object {
    "E0": 28,
  },
]
`;

exports[`An Octaves Test Suite should map - natural 1`] = `
Array [
  Object {
    "C-2": 0,
  },
  Object {
    "D-2": 2,
  },
  Object {
    "E-2": 4,
  },
  Object {
    "F-2": 5,
  },
  Object {
    "G-2": 7,
  },
  Object {
    "A-2": 9,
  },
  Object {
    "B-2": 11,
  },
  Object {
    "C-1": 12,
  },
  Object {
    "D-1": 14,
  },
  Object {
    "E-1": 16,
  },
  Object {
    "F-1": 17,
  },
  Object {
    "G-1": 19,
  },
  Object {
    "A-1": 21,
  },
  Object {
    "B-1": 23,
  },
  Object {
    "C0": 24,
  },
  Object {
    "D0": 26,
  },
  Object {
    "E0": 28,
  },
  Object {
    "F0": 29,
  },
  Object {
    "G0": 31,
  },
  Object {
    "A0": 33,
  },
  Object {
    "B0": 35,
  },
  Object {
    "C1": 36,
  },
  Object {
    "D1": 38,
  },
  Object {
    "E1": 40,
  },
  Object {
    "F1": 41,
  },
  Object {
    "G1": 43,
  },
  Object {
    "A1": 45,
  },
  Object {
    "B1": 47,
  },
  Object {
    "C2": 48,
  },
  Object {
    "D2": 50,
  },
  Object {
    "E2": 52,
  },
  Object {
    "F2": 53,
  },
  Object {
    "G2": 55,
  },
  Object {
    "A2": 57,
  },
  Object {
    "B2": 59,
  },
  Object {
    "C3": 60,
  },
  Object {
    "D3": 62,
  },
  Object {
    "E3": 64,
  },
  Object {
    "F3": 65,
  },
  Object {
    "G3": 67,
  },
  Object {
    "A3": 69,
  },
  Object {
    "B3": 71,
  },
  Object {
    "C4": 72,
  },
  Object {
    "D4": 74,
  },
  Object {
    "E4": 76,
  },
  Object {
    "F4": 77,
  },
  Object {
    "G4": 79,
  },
  Object {
    "A4": 81,
  },
  Object {
    "B4": 83,
  },
  Object {
    "C5": 84,
  },
  Object {
    "D5": 86,
  },
  Object {
    "E5": 88,
  },
  Object {
    "F5": 89,
  },
  Object {
    "G5": 91,
  },
  Object {
    "A5": 93,
  },
  Object {
    "B5": 95,
  },
  Object {
    "C6": 96,
  },
  Object {
    "D6": 98,
  },
  Object {
    "E6": 100,
  },
  Object {
    "F6": 101,
  },
  Object {
    "G6": 103,
  },
  Object {
    "A6": 105,
  },
  Object {
    "B6": 107,
  },
  Object {
    "C7": 108,
  },
  Object {
    "D7": 110,
  },
  Object {
    "E7": 112,
  },
  Object {
    "F7": 113,
  },
  Object {
    "G7": 115,
  },
  Object {
    "A7": 117,
  },
  Object {
    "B7": 119,
  },
  Object {
    "C8": 120,
  },
  Object {
    "D8": 122,
  },
  Object {
    "E8": 124,
  },
  Object {
    "F8": 125,
  },
  Object {
    "G8": 127,
  },
]
`;

exports[`An Octaves Test Suite should map - pentatonic 1`] = `
Array [
  Object {
    "Eb-2": 3,
  },
  Object {
    "F-2": 5,
  },
  Object {
    "G-2": 7,
  },
  Object {
    "Bb-2": 10,
  },
  Object {
    "C-1": 12,
  },
]
`;

exports[`An Octaves Test Suite should map - sharp 1`] = `
Array [
  Object {
    "E3": 64,
  },
  Object {
    "F#3": 66,
  },
  Object {
    "G#3": 68,
  },
  Object {
    "A3": 69,
  },
  Object {
    "B3": 71,
  },
  Object {
    "C#4": 73,
  },
  Object {
    "D#4": 75,
  },
  Object {
    "E4": 76,
  },
  Object {
    "F#4": 78,
  },
  Object {
    "G#4": 80,
  },
  Object {
    "A4": 81,
  },
  Object {
    "B4": 83,
  },
  Object {
    "C#5": 85,
  },
  Object {
    "D#5": 87,
  },
]
`;

exports[`An Octaves Test Suite should map 9m chord 1`] = `
Array [
  Object {
    "Ab4": 80,
  },
  Object {
    "B4": 83,
  },
  Object {
    "Eb5": 87,
  },
  Object {
    "Gb5": 90,
  },
  Object {
    "Bb5": 94,
  },
]
`;

exports[`An Octaves Test Suite should map M13 chord 1`] = `
Array [
  Object {
    "C4": 72,
  },
  Object {
    "E4": 76,
  },
  Object {
    "G4": 79,
  },
  Object {
    "B4": 83,
  },
  Object {
    "D5": 86,
  },
  Object {
    "A5": 93,
  },
  Object {
    "C6": 96,
  },
  Object {
    "E6": 100,
  },
  Object {
    "G6": 103,
  },
  Object {
    "B6": 107,
  },
  Object {
    "D7": 110,
  },
  Object {
    "A7": 117,
  },
]
`;
